created interfaces CarDao and DriverDao in package dao,

ShemsheyMaxim opened the pull request:
created classes CarDaoImpl and DriverDaoImpl in package dao,
created interfaces CarService and DriverService in package service,
created classes CarServiceImpl and DriverServiceImpl in package service,
added methods addCar and addDriver to class Storage

ShemsheyMaxim committed:
created interfaces CarDao and DriverDao in package dao,
created classes CarDaoImpl and DriverDaoImpl in package dao,
created interfaces CarService and DriverService in package service,
created classes CarServiceImpl and DriverServiceImpl in package service,
added methods addCar and addDriver to class Storage

ShemsheyMaxim committed:
fixed checkstyle

RomanDubovskyi commented on src/main/java/taxi/service/CarServiceImpl.java:
```diff
@@ -0,0 +1,68 @@
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).orElseThrow(() ->
+                new RuntimeException("Id don't found " + id));
```
```suggestion
                new RuntimeException("Car with id " + id + " not found"));
```

ShemsheyMaxim replied:
done

RomanDubovskyi commented on src/main/java/taxi/service/CarServiceImpl.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> result = new ArrayList<>();
+        for (Car car : carDao.getAll()) {
+            for (Driver driver : car.getDrivers()) {
+                if (driver.getId().equals(driverId)) {
+                    result.add(car);
+                    break;
+                }
+            }
+        }
```
move this logic to the dao layer, we should receive filtered data from dao

ShemsheyMaxim replied:
done

RomanDubovskyi commented on src/main/java/taxi/service/CarServiceImpl.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (!get(car.getId()).getDrivers().contains(driver)) {
```
we don't need this check, db will handle this situation and throw an exception if driver doesn't exist
and don't forget to call  the update method after

ShemsheyMaxim replied:
done

RomanDubovskyi commented on src/main/java/taxi/service/CarServiceImpl.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        get(car.getId()).getDrivers().removeIf(currentDriver ->
```
same here

ShemsheyMaxim replied:
done

ShemsheyMaxim committed:
Added method getAllByDriver to interface CarDao and CarDaoImpl,
deleted reduce check in methods addDriverToCar and removeDriverFromCar from class CarServiceImpl.

RomanDubovskyi commented on src/main/java/taxi/dao/CarDaoImpl.java:
```diff
@@ -0,0 +1,61 @@
+    public Optional<Car> get(Long id) {
+        return Optional.ofNullable(Storage.getCars().stream()
+                .filter(car -> car.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Car with id " + id + " not find")));
```
```suggestion
                .findFirst();
```
there is no point in returning optional if you throw an exception when the object is not found

RomanDubovskyi reviewed (approved):
nice, but remove throwing an exception in the get method on the dao layer before merge
